pkg/cmd/server/api/latest/helpers.go

```diff
@@ -158,3 +160,31 @@ func IsAdmissionPluginActivated(reader io.Reader, defaultValue bool) (bool, erro
 
 	return !activationConfig.Disable, nil
 }
+
+func ConvertOpenshiftAdmissionConfigToKubeAdmissionConfig(in map[string]configapi.AdmissionPluginConfig) (*apiserver.AdmissionConfiguration, error) {
+	ret := &apiserver.AdmissionConfiguration{}
+
+	for _, pluginName := range sets.StringKeySet(in).List() {
+		openshiftConfig := in[pluginName]
+
+		fmt.Printf("#### adding for %T\n", openshiftConfig.Configuration)
+		kubeConfig := apiserver.AdmissionPluginConfiguration{
+			Name: pluginName,
+			Path: openshiftConfig.Location,
+		}
+
+		if openshiftConfig.Configuration != nil {
+			configBytes, err := runtime.Encode(Codec, openshiftConfig.Configuration)
+			if err != nil {
+				return nil, err
```

Comment:
is this intended, when one fails, fail all? why not continue with others?

Reply:
> is this intended, when one fails, fail all? why not continue with others?

We don't ever want to construct a partial admission chain since that could leave us unprotected by the admission validation we expect.  Missing config could produce that problem.

Reply:
I agree completely. Though I wonder, for that if we should have a validation check (if we dont have one already) to make sure final admission plugin chain (that are going to be enabled) is secure instead of relying on this. That'd also help providing warning  to admins in case they have not configured a secure chain.

Reply:
> I agree completely. Though I wonder, for that if we should have a validation check (if we dont have one already) to make sure final admission plugin chain (that are going to be enabled) is secure instead of relying on this. That'd also help providing warning to admins in case they have not configured a secure chain.

We leave the final determination of on and off to the cluster-admin.  If they intentionally make it insecure by disablign plugins (without producing errors in the configuration), then we allow them to do that.